external_status_checks.go

```diff
@@ -197,3 +197,31 @@ func (s *ExternalStatusChecksService) UpdateExternalStatusCheck(pid interface{},
 
 	return s.client.Do(req, nil)
 }
+
+// RetryFailedStatusCheckForAMergeRequestOptions represents the available
+// RetryFailedStatusCheckForAMergeRequest() options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/status_checks.html#retry-failed-status-check-for-a-merge-request
+type RetryFailedStatusCheckForAMergeRequestOptions struct {
+	ExternalStatusCheckID *int `url:"external_status_check_id,omitempty" json:"external_status_check_id,omitempty"`
+}
+
+// UpdateExternalStatusCheck updates an external status check.
+//
+// Gitlab API docs:
+// https://docs.gitlab.com/ee/api/status_checks.html#retry-failed-status-check-for-a-merge-request
+func (s *ExternalStatusChecksService) RetryFailedStatusCheckForAMergeRequest(pid interface{}, mergeRequest int, opt *RetryFailedStatusCheckForAMergeRequestOptions, options ...RequestOptionFunc) (*Response, error) {
```

Comment:
All the params that are part of the URL are passed as function parameters. The option structs are only used for sending options to the API which in this case is not applicable as all params are art of the URL and not request body is expected.